src/__tests__/index.test.ts

```diff
@@ -0,0 +1,512 @@
+import util from 'util';
+import assert from 'assert';
+import { stub, SinonStub } from 'sinon';
+import oracledb, { Connection, PoolAttributes } from 'oracledb';
+import OracleWrapper from '../index';
+import { Logger, OracleDBOptions } from '../types';
+
+describe('OracleWrapper', function () {
+  let wrapper: OracleWrapper;
+  let getConnectionStub: SinonStub;
+  let createPoolStub: SinonStub;
+  let connectionPool: Record<string, SinonStub>;
+  let dbConnection: Record<string, SinonStub>;
+  let queryResult: Record<string, Record<string, SinonStub>>;
+  const credentials = {
+    host: '127.0.0.1',
+    port: '1521',
+    sid: 'ORCTEST',
+    user: 'SYSADMIN',
+    password: 'oracledbtest',
+  };
+  let options: OracleDBOptions;
+  let testLogger: Record<string, SinonStub>;
+
+  beforeEach(function () {
+    testLogger = {
+      log: stub(),
+      error: stub(),
+      warn: stub(),
+      info: stub(),
+      debug: stub(),
+      verbose: stub(),
+    };
+    connectionPool = {
+      close: stub(),
+    };
+    dbConnection = {
+      execute: stub(),
+      release: stub(),
+    };
+    queryResult = {
+      resultSet: {
+        getRows: stub(),
+        close: stub(),
+      },
+    };
+    options = {
+      alias: 'TestDB1',
+      logger: testLogger as unknown as Logger,
+    };
```

Comment:
I don't think we could split these out, since we need the mocks to be instantiated within the scope of mocha, so that they'll be reset as expected after each test.

Reply:
Ah right. Jest lets you do that - for some reason I thought Mocha did too.